pkg/secret/secrets.go

```diff
@@ -13,34 +13,36 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const JwksSecretKey = "jwks"
+const JwkSecretKey = "jwk"
 const SecretLabelKey = "type"
 const SecretLabelType = "jwker.nais.io"
 
-func ExtractJWKS(sec corev1.Secret) (jose.JSONWebKeySet, error) {
-	jwks := jose.JSONWebKeySet{}
-	err := json.Unmarshal(sec.Data[JwksSecretKey], &jwks)
-	return jwks, err
-}
-
-func ExtractJWK(sec corev1.Secret) (*jose.JSONWebKey, error) {
-	jwks, err := ExtractJWKS(sec)
-	if err != nil {
-		return nil, err
-	}
+func FirstJWK(jwks jose.JSONWebKeySet) (*jose.JSONWebKey, error) {
 	keysLen := len(jwks.Keys)
 	if keysLen != 1 {
 		return nil, fmt.Errorf("secret has %d keys, expecting exactly 1", keysLen)
 	}
 	return &jwks.Keys[0], nil
 }
 
+func ExtractJWK(sec corev1.Secret) (*jose.JSONWebKey, error) {
+	jwk := jose.JSONWebKey{}
+	err := json.Unmarshal(sec.Data[JwkSecretKey], &jwk)
+	return &jwk, err
```

Comment:
Should have a case where you return nil, it's best practice when you return an error.